compiler/rustc_hir/src/def.rs

```diff
@@ -47,33 +49,40 @@ pub enum NonMacroAttrKind {
     Registered,
 }
 
+/// What kind of definition something is; e.g., `mod` vs `struct`.
 #[derive(Clone, Copy, PartialEq, Eq, Encodable, Decodable, Hash, Debug)]
 #[derive(HashStable_Generic)]
 pub enum DefKind {
     // Type namespace
     Mod,
-    /// Refers to the struct itself, `DefKind::Ctor` refers to its constructor if it exists.
+    /// Refers to the struct itself, [`DefKind::Ctor`] refers to its constructor if it exists.
     Struct,
     Union,
     Enum,
-    /// Refers to the variant itself, `DefKind::Ctor` refers to its constructor if it exists.
+    /// Refers to the variant itself, [`DefKind::Ctor`] refers to its constructor if it exists.
     Variant,
     Trait,
-    /// `type Foo = Bar;`
+    /// Type alias: `type Foo = Bar;`
     TyAlias,
     ForeignTy,
+    /// Trait alias: `type IntIterator = Iterator<Item = i32>;`
     TraitAlias,
+    /// Associated type: `trait MyTrait { type Assoc; }`
     AssocTy,
+    /// Type parameter: the `T` in `Vec<T>`
     TyParam,
 
     // Value namespace
     Fn,
     Const,
+    /// Const-generic type parameter.
```

Comment:
```suggestion
    /// Constant generic parameter `struct Foo<const S: usize> { ... }`.
```

Reply:
(replaced `S` with `N`)

```suggestion
    /// Constant generic parameter `struct Foo<const N: usize> { ... }`.
```